docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,861 @@
+
+The installer is still used to generate the ignition files and monitor the installation process.
+
+This provides a greater flexibility at the cost of a more explicit and interactive process.
+
+
+## Prerequisites
+
+The requirements for UPI are broadly similar to the [ones for OpenStack IPI][ipi-reqs]:
+
+[ipi-reqs]: ./openstack#openstack-requirements
+
+- OpenStack account with `clouds.yaml`
+  - This document will use a cloud called `openstack`
+- Nova flavors
+  - This document will use `m1.xlarge` for masters and `m1.large` for workers
+- An external subnet you want to use for floating IP addresses
+  - This document will use `external`
+- The `openshift-install` binary in your `$PATH`
+- The [Red Hat CoreOS][rhcos] image in Glance
+  - This document will use the `rhcos` image name
+- A subnet range for the Nova servers / OpenShift Nodes
+  - This range must not conflict with your existing network
+  - This document will use `192.0.2.0/24`
+- A cluster name you will want to use
+  - This document will use `openshift`
+- A base domain
+  - This document will use `example.com`
+- OpenShift Pull Secret
+- DNS infrastructure you can configure (you'll need to add two records there)
+
+[rhcos]: https://www.openshift.com/learn/coreos/
+
+The OpenShift API URL will be generated from the cluster name and base domain. E.g.: `https://api.openshift.example.com:6443/`
+
+
+You can validate most of the above by running the following commands:
+
+```sh
+$ export OS_CLOUD=openstack
+$ openstack image show rhcos
+$ openstack network show external
+$ openstack flavor show m1.xlarge
+$ openstack flavor show m1.large
+$ which openshift-install
+```
+
+They should all succeed.
+
+
+## OpenShift Configuration Directory
+
+All the configuration files, logs and installation state is kept in a single directory:
+
+```sh
+$ mkdir -p openstack-upi
+$ cd openstack-upi
+```
+
+## API and Ingress Floating IP Addresses
+
+```sh
+$ openstack floating ip create --description "OpenShift API" <external>
+=> 203.0.113.23
+$ openstack floating ip create --description "OpenShift Ingress" <external>
+=> 203.0.113.19
+```
+
+**NOTE**: throughout this document, we will use `203.0.113.23` as the public IP address for the OpenShift API endpoint and `203.0.113.19` as the public IP for the ingress (`*.apps`) endpoint.
+
+## Install Config
+
+Run the `create install-config` subcommand and fill in the desired entries:
+
+```sh
+$ openshift-install create install-config
+? SSH Public Key </home/user/.ssh/id_rsa>
+? Platform <openstack>
+? Cloud <openstack>
+? ExternalNetwork <external>
+? APIFloatingIPAddress <203.0.113.23>
+? FlavorName <m1.xlarge>
+? Base Domain <example.com>
+? Cluster Name <openshift>
+```
+
+Most of these are self-explanatoray. `Cloud` is the cloud name in your `clouds.yaml` i.e. what's set as your `OS_CLOUD` environment variable.
```

Comment:
I must confess that it is still not entirely clear to me why `Base Domain` is required here, and how it is used by the installer.

Do you think we can add a very short explanation without cluttering the text?

Something like:
```
The fully qualified domain name with which the API interface will expect to the called, and the default name with which OpenShift will expose newly created applications.
```

WDYT?